katharsis-jpa/src/main/java/io/katharsis/jpa/internal/meta/impl/AbstractMetaEntityAttributeImpl.java

```diff
@@ -0,0 +1,126 @@
+package io.katharsis.jpa.internal.meta.impl;
+
+import java.beans.PropertyDescriptor;
+import java.lang.reflect.Field;
+
+import javax.persistence.ElementCollection;
+import javax.persistence.FetchType;
+import javax.persistence.ManyToMany;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.persistence.OneToOne;
+import javax.persistence.Version;
+
+import io.katharsis.jpa.internal.meta.MetaAttribute;
+import io.katharsis.jpa.internal.meta.MetaEntity;
+import io.katharsis.jpa.internal.meta.MetaLookup;
+import io.katharsis.jpa.internal.meta.MetaType;
+
+public class AbstractMetaEntityAttributeImpl extends MetaAttributeImpl {
+
+	protected Field field;
+	private boolean derived;
+
+	private boolean lazy = false;
+	private String mappedBy = null;
+	private MetaAttribute oppositeAttr;
+	private boolean version = false;
+
+	public AbstractMetaEntityAttributeImpl(MetaDataObjectImpl parent, PropertyDescriptor desc) {
+		super(parent, desc);
+
+		field = getField(parent, desc);
+		if (field != null) {
+			readAnnotations(field);
+		} else {
+			derived = true;
+		}
+	}
+
+	private static Field getField(MetaDataObjectImpl parent, PropertyDescriptor desc) {
+		try {
+			Field field = parent.getImplementationClass().getDeclaredField(desc.getName());
+			field.setAccessible(true);
+			return field;
+		} catch (NoSuchFieldException e) { // NOSONAR
+			return null;
+		}
+	}
+
+	private void readAnnotations(Field field) { // NO SONAR
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) The Cyclomatic Complexity of this method "readAnnotations" is 13 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)